src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -89,8 +101,9 @@ public static bool IsRequestKeyValid(string value)
             }
             try
             {
-                byte[] data = Convert.FromBase64String(value);
-                return data.Length == 16;
+                Span<byte> temp = stackalloc byte[16];
+                var success = Convert.TryFromBase64String(value, temp, out var written);
+                return written == 16 && success;
             }
             catch (Exception)
```

Comment:
Is there any chance for an exception left in this code-path?

`Convert.TryFromBase64String` throws if `value`-argument is `null`, but this case is already eliminated in L98 here.
So you could remove the `try-catch`, resulting in smaller code.

Actually I would write this method as
```c#
public static bool IsRequestKeyValid(string value)
{
    ReadOnlySpan<char> chars = value.AsSpan();

    if (chars.IsEmpty)
    {
        return false;
    }

    Span<byte> temp = stackalloc byte[16];
    var success = Convert.TryFromBase64Chars(chars, temp, out int written);

    return success && written == 16;
}
```

Maybe: to get even more perf, one could skip the base64-decoding step, and just check if the `chars.Length == 16` and if the `chars` are in the base64-alphabet, so no need to perform the actual decoding. This step could also be vectorized. But maybe that's a bit of over-engineering.

Reply:
If `written == 16`, so for base64 the input-length must be `24`. Hence one could write
```c#
ReadOnlySpan<char> chars = value.AsSpan();

if (chars.Length != 24)
{
    return false;
}
```
So the entire decoding-step is omitted if the result (written) can't be 16.

Reply:
> Maybe: to get even more perf, one could skip the base64-decoding step, and just check if the `chars.Length == 16` and if the `chars` are in the base64-alphabet, so no need to perform the actual decoding.

I'm assuming you meant `chars.Length == 24`. Can't do this because a 17 and 18 byte string will also be 24 bytes when encoded. Unless you also want to check for the trailing "==", but I think at this point we're trying too hard.

I'm going to stick with `Convert.TryFromBase64Chars` for now, but I'll look at the other feedback :)

Reply:
> I'm assuming you meant `chars.Length == 24`

Yep, typed wrong in the first comment.

> Can't do this because a 17 and 18 byte string will also be 24 bytes when encoded.

Maybe I missunderstand something now, but the string is encoded and we try to decode it, then check if the decoded data (byte) is of size 16. 
So the other way round this means, 16 data bytes -> 24 base64 encoded utf8-bytes.
And for base64 the encoded length is always a multiple of 4.

Thus if `chars.Length != 24` we know that the decoded data won't be 16 bytes, so we can short-circuit the method and return `false`. No need to try to decode the input.

If `chars.Length == 24` we can try to decode, then check if `written == 16`.

Ahhhhh, now I see that in my previous comment it may look like the entire method. It's just a fragment. Sorry.

So I mean:
```c#
public static bool IsRequestKeyValid(string value)
{
    ReadOnlySpan<char> chars = value.AsSpan();

    if (chars.Length != 24)
    {
        return false;
    }

    Span<byte> temp = stackalloc byte[16];
    var success = Convert.TryFromBase64Chars(chars, temp, out int written);

    return written == 16 && success;
}
```

If you want I could submit a PR for the vectorized version (next week).